frontend/src/App.js

```diff
@@ -9,26 +9,51 @@ import Landing from "pages/landing/Landing";
 import ProductOverview from "pages/product-overview/ProductOverview";
 import Search from "pages/search/Search";
 import "./App.css";
+import Login from "pages/login/Login";
+import {UserContext} from "context/UserContext";
+import {getCurrentUser} from "api/User";
+import Register from "pages/register/Register";
 
 function App() {
+  const [user, setUser] = useState(JSON.parse(localStorage.getItem("user")));
+  const value = useMemo(() => ({user, setUser}), [user, setUser]);
+
+  useEffect(() => {
+    getCurrentUser(localStorage.getItem("token"))
```

Comment:
Why is this call necessary, isn't the user already set if it's present in the local storage?